packages/py-api/py_api/controllers/hackathon/verification_controller.py

```diff
@@ -5,38 +5,76 @@
-            return JSONResponse(status_code=401)
+    async def verify_participants(cls, jwt_token: str) -> JSONResponse | Dict[str, str]:
+        result = JWTFunctionality.decode_token(jwt_token)
+
+        if isinstance(result, dict):
+            decoded_token = result
+        else:
+            return JSONResponse(content=result[0], status_code=result[1])
 
         team = TeamFunctionality.fetch_team(
             team_name=decoded_token.get("team_name"),
         )
-
         if not team:
-            return JSONResponse(content={"message": "Cannot verify admin as such team doesn't exist"}, status_code=404)
-
-        # TODO: verify expiration
+            return JSONResponse(
+                content={
+                    "message": "Cannot verify participant as such team doesn't exist",
+                },
+                status_code=404,
+            )
 
-        verified_admin = ParticipantsFunctionality.update_participant(
-            id=team.team_members[0], participant=UpdateParticipant(
+        verified_participant = ParticipantsFunctionality.update_participant(
+            object_id=str(decoded_token.get("sub")), participant=UpdateParticipant(
                 is_verified=True,
             ),
         )
+        if not verified_participant:
+            return JSONResponse(
+                content={
+                    "message": "Something went wrong updating participant document",
+                },
+                status_code=500,
+            )
 
-        if not verified_admin:
-            return JSONResponse(content={"message": "Something went wrong updating admin document"}, status_code=500)
+        if team.is_verified is not True:
+            team.is_verified = True
+            verified_team = TeamFunctionality.update_team_query_using_dump(
+                team_payload=team.model_dump(),
+            )
 
-        # TODO: add is_verified property to teams so that you can later delete expired teams upon admin clean ups
-        # TODO: send email with jwt link for appending participants to now verified team
+            if not verified_team:
+                return JSONResponse(
+                    content={
+                        "message": "Something went wrong updating team document",
+                    },
+                    status_code=500,
+                )
 
-        return {"message": "Admin was successfully verified"}
+        if team.team_type == team.team_type.NORMAL:
+            try:
+                jwt_token = JWTFunctionality.create_jwt_token(
+                    team_name=team.team_name, is_invite=True,
+                )
+                await send_mail(
+                    verified_participant.get("email"), "Test",
+                    f"Token: {JWTFunctionality.get_verification_link(jwt_token)}",
+                )
+                # background_tasks = BackgroundTasks()
+                # background_tasks.add_task(background_send_mail, verified_participant.get("email"), "Test",
```

Comment:
This for some reason does not work, so I opted for just awaiting the send_email func. If you have ideas on how to send the email as a background task, I am happy to hear them.

Reply:
I am using the asynio.create_task but there should be a better way. Can you advise me on that?

Reply:
Look at how the analytics endpoint works and sends a task in the background.